### PR TITLE
chore: Update "Allow Support Access" to toggle and move above send button

### PR DESCRIPTION
**Pull request by monicakh**
## What kind of change does this PR introduce?

1. Converted "Allow Support Access" from a checkbox to a toggle switch
2. Repositioned the "Allow Support Access" control above the "Send" button

### CODE REVIEW

**Comment by CarelFdeWaal** [comment]
> Thanks @monicakh. I was not aware of this PR. Here is the context: [Notion task](https://www.notion.so/Ticket-View-Request-permission-to-use-AI-13e5004b775f8071afa7da806e11e03a?pvs=8&n=github_linkback). Perhaps we incorporate the new wording(https://github.com/supabase/supabase/pull/30540/files) also into this design.

@monicakh see discussion: https://supabase.slack.com/archives/C0429V78ACX/p1732687370649219
(Meanwhile it should not block other PR)

**Comment by MildTomato** [comment]
Some things I noticed

- we are storing in the form the state, but also have a useState. so there's some dupication
- using deprecated `<Toggle />` component
- I've restored the old style of submit button as it looked a bit odd to me

File: apps/studio/components/interfaces/Support/SupportFormV2.tsx [hunk 1]
```diff
@@ -76,6 +79,23 @@ export const SupportFormV2 = ({ setSentCategory, setSelectedProject }: SupportFo
...
+
+  const onToggleOptIn = () => {
+    const value = !snap.isOptedInTelemetry ? 'true' : 'false'
+    snap.setIsOptedInTelemetry(value === 'true')
+    if (value === 'false') sendReset()
+    setSupportAccess(value === 'true')
+    form.setValue('allowSupportAccess', value === 'true')
+  }
+
+  const Divider = () => {
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Support/SupportFormV2.tsx:95]: `<Seperator/>` component

**Comment by MildTomato** [comment]
updated styling a bit

Screenshot 2024-11-27 at 20 23 05

**Review by CarelFdeWaal — APPROVED** [review]

**Review by staaldraad — APPROVED** [review]

**Review by MildTomato — CHANGES_REQUESTED** [review]

File: apps/studio/components/interfaces/Support/SupportFormV2.tsx [hunk 2]
```diff
@@ -705,27 +679,83 @@ export const SupportFormV2 = ({ setSentCategory, setSelectedProject }: SupportFo
...
+            <div className="flex flex-col gap-5">
+              {['Problem', 'Database_unresponsive', 'Performance'].includes(category) && (
+                <>
+                  <FormField_Shadcn_
+                    name="allowSupportAccess"
+                    control={form.control}
+                    render={({ field }) => (
+                      <FormItemLayout
+                        className="px-6"
+                        layout="flex"
```
[the remark this answers is not in the document]
Comment (MildTomato) [reply apps/studio/components/interfaces/Support/SupportFormV2.tsx:692]: for more context, the `label` on `<FormItemLayout />` adds a `<label />` element, and it adds a name attribute to it that then allows a click on it to toggle the switch.

[the remark this answers is not in the document]
Comment (monicakh) [reply apps/studio/components/interfaces/Support/SupportFormV2.tsx]: Thanks @MildTomato, I've updated the code as suggested.

Also, here’s how the design looks now:

image

**Review by MildTomato — APPROVED** [review]

**Pull request merged**